tests/unit/s2n_session_ticket_test.c

```diff
@@ -1478,6 +1478,178 @@ int main(int argc, char **argv)
         EXPECT_EQUAL(client->ticket_lifetime_hint, 0);
     }
 
+    /* Test: Server disables legacy tickets */
```

Comment:
We should also cover the case where legacy tickets are disabled on the server but the client sends one anyway

Reply:
What exactly would this look like? Do you mean if the client sends a TLS1.3 ticket? They can't send a TLS1.2 ticket because the server isn't producing them.

Reply:
I mean if the client sends a TLS1.2 ticket, not a TLS1.3 ticket. Another server could be producing TLS1.2 tickets with the same STEK, or the server could have produced a TLS1.2 ticket previously. Regardless of how the ticket exists, the server should reject it.